src/main/app/case/case-api.ts

```diff
@@ -98,33 +99,31 @@ export class CaseApi {
   }
 
   private getPriorityUserCase(userCases: CcdV1Response[] | false): CaseWithId | false {
-    const preSubmissionJointStates = [
-      State.AwaitingApplicant1Response,
-      State.AwaitingApplicant2Response,
-      State.Applicant2Approved,
-    ];
+    // If more than one userCase, find the priority one (e.g. Submitted > AwaitingPayment > Draft):
     if (userCases && userCases.length > 1) {
-      const submittedUserCase = userCases.find(
-        userCase => ![State.Draft, ...preSubmissionJointStates].includes(userCase.state)
-      );
-      const preSubmittedUserCase = userCases.find(userCase => preSubmissionJointStates.includes(userCase.state));
-      const priorityUserCase = submittedUserCase || preSubmittedUserCase;
-      if (priorityUserCase) {
-        return {
-          ...fromApiFormat(priorityUserCase.case_data),
-          id: priorityUserCase.id.toString(),
-          state: priorityUserCase.state,
-        };
+      const submittedUserCase = userCases.find(userCase => !preSubmittedStatePrioritySequence.includes(userCase.state));
+      if (submittedUserCase) {
+        return convertCcdV1ResponseToCaseWithId(submittedUserCase);
       }
+      // If we have two Draft cases, this method is safe in finding the latest case because the [0] element will be the latest case.
+      // Otherwise, the highest priority pre-submitted case will be chosen (the one with the highest index):
+      return convertCcdV1ResponseToCaseWithId(this.getHighestPriorityPreSubmissionCases(userCases)[0]);
     }
+    // Else, return the only case we have (or false):
     return this.getLatestUserCase(userCases);
   }
 
+  private getHighestPriorityPreSubmissionCases(userCases: CcdV1Response[]): CcdV1Response[] {
+    const stateIndexArr = userCases.map(userCase => preSubmittedStatePrioritySequence.indexOf(userCase.state));
+    const highestPriorityStateIndex: number = Math.max(...stateIndexArr);
```

Comment:
maybe we could add this as a function in the state-sequence class, as we already have before and after methods there.